### PR TITLE
Rename common.platform to common.os and split among windows, linux and unix utils

### LINKED ISSUE #7411: Split common/platform  module on conda to common/os/<os>
This is a refactor of the code to split utilities that refer to specific operating systems

https://github.com/conda/conda/blob/master/conda/common/platform.py

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes #7411

### CODE REVIEW

**Comment by goanpeca** [comment]
Is this what you had in mind for the split @kalefranz  ?

File: conda/common/os/__init__.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from ..os.linux import linux_get_libc_version
+from ..os.unix import get_free_space_on_unix, is_admin_on_unix
+from ..os.windows import get_free_space_on_windows, is_admin_on_windows
+from ..compat import on_win
+
+
+log = getLogger(__name__)
+
+
+linux_get_libc_version
```
Comment (kalefranz) [inline conda/common/os/__init__.py]: We shouldn't be importing from all platforms all the time here.  Should be more like

```python
if on_win:
    from .windows import is_admin_on_windows as is_admin
else:
    from .unix import is_admin_on_unix as is_admin
```

We don't need the `linux_get_libc_version` import at all in this module I don't think.

File: conda/common/os/linux.py [hunk 2]
```diff
@@ -4,45 +4,22 @@
...
 from genericpath import exists
 from glob import glob
 from logging import getLogger
-import os
 import sys
 
-from .compat import iteritems, on_win
-from .._vendor.auxlib.decorators import memoize
+from ..._vendor.auxlib.decorators import memoize
+from ...compat import iteritems
```
Comment (kalefranz) [inline conda/common/os/linux.py]: This should just be 

    from .compat import iteritems

File: conda/common/os/windows.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from logging import getLogger
+
+from ...compat import on_win
```
Comment (kalefranz) [inline conda/common/os/windows.py]: Don't use `conda/compat.py`.  Use `conda/common/compat.py`.

    from .compat import on_win

**Pull request merged**